src/order/locally_finite.lean

```diff
@@ -482,6 +482,246 @@ locally_finite_order.of_Icc' (α × β)
 
 end preorder
 
+/-! #### `with_top`, `with_bot` -/
+
+section order_top
+open with_top
+variables (α) [order_top α] [locally_finite_order α]
+
+instance : locally_finite_order (with_top α) :=
+{ finset_Icc := λ a b, match a, b with
+    |        ⊤,        ⊤ := {⊤}
+    |        ⊤, (some b) := ∅
+    | (some a),        ⊤ := cons (⊤ : with_top α) ((Ici a).map embedding.some)
+                              (λ h, let ⟨x, _, hx⟩ := mem_map.1 h in coe_ne_top hx)
+    | (some a), (some b) := (Icc a b).map embedding.some
+    end,
+  finset_Ico := λ a b, match a with
```

Comment:
You can do a double pattern match here too, and match ` ⊤, _` for the first case